src/main/java/hudson/model/ExternalJob.java

```diff
@@ -76,7 +78,11 @@ public synchronized ExternalRun newBuild() throws IOException {
         long timeSinceLast = System.currentTimeMillis() - lastBuildStartTime;
         if (timeSinceLast < 1000) {
             try {
-                Thread.sleep(1000 - timeSinceLast);
+                while(timeSinceLast > 0){
+                    wait();
```

Comment:
:bug: huh? What do you think is going to interrupt this?

I suppose you saw some FB warning about sleeping not in a loop—fine. But that is fixed by simply calling `sleep` in a loop.

Or, more simply, by updating to 1.609.x and deleting this whole hack, which is obsolete as of 1.597 IIRC.